zoe-cli/src/commands/main.kt

```diff
@@ -98,31 +76,27 @@ class ZoeCommandLine : CliktCommand(name = "zoe") {
         )
     }
 
+    fun buildCliContext() = CliContext(
+        home = home,
+        runner = runner,
+        cluster = cluster,
+        configDir = configDir,
+        env = env,
+        term = term,
+        version = loadFileFromResources("version.json")?.toJsonNode()?.get("projectVersion")?.asText()
+            ?: kotlin.run {
+                logger.warn("Couldn't find version from resources. Using 'latest'")
+                "latest"
+            }
+    )
+
     override fun run() {
         LogManager.getRootLogger().level = when {
             silent -> Level.ERROR
             verbosity <= 1 -> Level.INFO
             verbosity == 2 -> Level.DEBUG
             else -> Level.ALL
         }
-
-        loadKoinModules(
```

Comment:
@wlezzar I moved the koin module definition out of the zoe main Command  and moved it into the main entry point of the whole cli. I think this is better place as there the whole koin lifecycle is already controlled + it is more clear in which order modules can be defined

Reply:
Hi @mgmeiner . I took a look at the PR. I see your point of moving the koin modules loading in the same place where koin is instantiated.

However, I feel this is not compatible with the philosophy of Clikt and the application: basically, as the Clikt commands are independently nested, each should be able to independently create its own beans and modules. Currently, this is achievable by making the Clikt command instantiate a koin module itself and this becomes available to the child commands (like the main CliktCommand is currently doing). If we make all the modules be loaded at the top level we will need to create more links and coupling across the commands themselves. Makes sense? what do you think?

Reply:
@wlezzar Ok i get your point but the problem in this case is the order in which the koin modules are loaded because we need to override the EnvConfig bean and so this must happen after the main module is loaded and if this configuration logic happens during "run phase" of the command its quite hard to intercept it. 

One option  would be to pass the additional koin modules as constructor parameter or provide a configuration function on ZoeCommandLine which passes additional modules. Latter option might be more readable and clear as Clikt does it the same way. 

Like:
```
val command =
    ZoeCommandLine()
         .subcommands(
              topicsCommand(),
              schemasCommand(),
              groupsCommands(),
              configCommands(),
              lambdaCommands(),
              versionCommands()
         )
         .context(customizeContext)
         .additionalKoinModules(additionalModules)
```
Another solution could be to pass a simple callback function which executes between configuration and actual "run".
Maybe you have a better solution in mind?

Reply:
Ok I completely see your point now.

I like the callback solution: I'm thinking of the following: create an `EnvConfigCustomizer` that extends `(EnvConfig) -> EnvConfig`. Then [in this block](https://github.com/adevinta/zoe/blob/master/zoe-cli/src/commands/main.kt#L180-L185) where we load the `EnvConfig`, we can check if such a bean exist in the module. If it does exist, it gets executed as the final step there. This `EnvConfigCustomizer` will be injected in the tests to override the configs.
What do you think of this a solution?

I also agree that there is something to clarify / fix in the koin modules loading but I suggest tackling this in a different PR / issue. What do you think?

(ps. sorry for the delay, I'm fully back now)